modules/apps/depot/depot-web/src/main/java/com/liferay/depot/web/internal/servlet/taglib/clay/DepotEntryVerticalCard.java

```diff
@@ -75,11 +74,6 @@ public DepotEntryVerticalCard(
 		return depotActionDropdownItemsProvider.getActionDropdownItems();
 	}
 
-	@Override
-	public String getDefaultEventHandler() {
-		return DepotAdminWebKeys.DEPOT_ENTRY_DROPDOWN_DEFAULT_EVENT_HANDLER;
-	}
-
```

Comment:
Besides here, `DepotAdminWebKeys.DEPOT_ENTRY_DROPDOWN_DEFAULT_EVENT_HANDLER` is used only in two `<clay:dropdown-actions>`.  Looking [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld#L863), `<clay:dropdown-actions>` also has `propsTransformer` attribute. If we can replace those usages, we can clean up everything related to default event handler, like delete `DepotEntryDropdownDefaultEventHandler.es.js` file and the Java constant.

Reply:
Sounds like a good idea.

Reply:
One thing I don't really like is that  [`DepotEntryDropdownDefaultEventHandler`](https://github.com/liferay/liferay-portal/blob/826c5123ca36f821eba3406147aff60ff3de9ef0/modules/apps/depot/depot-web/src/main/resources/META-INF/resources/js/DepotEntryDropdownDefaultEventHandler.es.js), and the "new" [`DepotEntryVerticalCardPropsTransformer`](https://github.com/liferay-frontend/liferay-portal/blob/0a1799a69a67b4381c2c0009aa8dbe50407a8a52/modules/apps/depot/depot-web/src/main/resources/META-INF/resources/js/DepotEntryVerticalCardPropsTransformer.js) are identical, *but* they are used by two components that have the "same" property (i.e. that prop has the same purpose) but with a different name:
[`DropDownWithItems`](https://github.com/liferay/clay/blob/e0a84fb91adebbf8005fce3bcb15abdb24ebd12e/packages/clay-drop-down/src/DropDownWithItems.tsx#L296) uses `items`, but [`CardWithInfo`](https://github.com/liferay/clay/blob/master/packages/clay-card/src/CardWithInfo.tsx#L108) uses `actions`.

Of course, we can do something like this:

```js
import {openModal} from 'frontend-js-web';
import confirmDepotEntryDeletion from './confirmDepotEntryDeletion.es';

const ACTIONS = {
	deleteDepotEntry(itemData) {
		if (confirmDepotEntryDeletion()) {
			submitForm(document.hrefFm, itemData.deleteDepotEntryURL);
		}
	},

	permissionsDepotEntry(itemData) {
		openModal({
			title: Liferay.Language.get('permissions'),
			url: itemData.permissionsDepotEntryURL,
		});
	},
};

function mapActionsOrItems(item) {
	return {
		...item,
		onClick(event) {
			const action = item.data?.action;

			if (action) {
				event.preventDefault();

				ACTIONS[action](item.data);
			}
		},
	};
}

export default function propsTransformer({actions, items, ...otherProps}) {
	const prop = actions ? 'actions' : items ? 'items' : null;

	return {
		...otherProps,
		[prop]: actions ? actions.map(mapActionsOrItems) : items.map(mapActionsOrItems),
	};
}
```

But it feels like a hack and fragile.

Do you guys have any idea how to improve that? (/cc @wincent ) 

(Opening an issue in clay to rename the prop in one of the two components, sending a PR, waiting for a release and update in DXP and updating the usages doesn't seen like an option)

Reply:
I guess we could do something like this too

```js
export default function propsTransformer({actions, items, ...otherProps}) {
	const result = {
		...otherProps,
	};

	if (actions) {
		result.actions = actions.map(mapActionsOrItems);
	}
	else if (items) {
		result.items = items.map(mapActionsOrItems);
	}

	return result;
}
```

Reply:
So that's what I chose to go with in [
171f3f5](https://github.com/liferay-frontend/liferay-portal/pull/495/commits/171f3f5304fdc7f31f90bb37c09aec30661d0310)

Reply:
If we're golfing here, technically speaking you don't need to make a copy of `otherProps` and name it `result`, because `otherProps` is _already_ a new object that you own and are free to mutate.

```js
export default function propsTransformer({actions, items, ...props}) {
    if (actions) {
        props.actions = actions.map(map);
    }
    else if (items) {
        props.items = items.map(map);
    }

    return props;
}
```

I don't know if you want to check against being improperly called (ie. with `actions` _and_ `items`) by throwing or logging an error. 🤷

Reply:
@wincent thanks, that's nicer.

> I don't know if you want to check against being improperly called (ie. with actions and items) by throwing or logging an error. :shrug:

sounds like a good idea

Reply:
@julien I ended up being late for this, I think I missed the Github notification 🤷‍♂️.

I think they are different contexts, keeping the same name of `items` for `ClayCardWithInfo` may not mean much, but` actions` adds more context to what to expect from this API, in my view and I think that also when we developed this, `items` in `ClayDropDownWithItems` makes sense because it is the main API for the component. So I'm not sure if it would be a good idea to keep the same API name on both components. Does that make sense to you?

Just remembering that we cannot change the name of the APIs now, we would have to maintain compatibility in any way or remove it only in v4 if we reach an agreement.

Reply:
@matuzalemsteles no problem we can keep the name `actions` for Cards and `items` for dropdowns.
 
> Just remembering that we cannot change the name of the APIs now,

Yes I know, I think that's also one thing we shouldn't have done between clay 2.x and clay 3.x (on the JS side) to make the replacements easier, but we can deal with it.